### ISSUE TITLE
git state does not take into account ssh config file

### ISSUE THREAD

**Issue by gidantribal** [saltstack/salt#32685]
Hello,

I want to use port 443 for targeting with ssh over https my repository. If I use ssh command manually I see it correclty uses port 433, but if I use the salt git state I am keeping using port 22.

**Comment by gidantribal** [comment]
```
          ID: git@altssh.bitbucket.org:443/agiambruno/dokuwiki
    Function: git.latest
      Result: False
     Comment: Failed to check remote refs: Unable to authenticate using identity file:

              ssh: connect to host altssh.bitbucket.org port 22: Connection refused
              fatal: Could not read from remote repository.

              Please make sure you have the correct access rights
              and the repository exists.
     Started: 18:37:46.929310
    Duration: 1046.393 ms
     Changes:

```

**Comment by gidantribal** [comment]
As a proof I tried to rungit manually (with state.cmd) and it works flawlessly (the uncommented block):

```
checkout-dokuwiki-repository:
  cmd.run:
    - name: git clone git@bitbucket.org:user/dokuwiki.git /var/www/html/wiki
    - user: root
    - require:
      - /root/.ssh/id_rsa
      - /root/.ssh/id_rsa.pub
      - /root/.ssh/known_hosts
      - /root/.ssh/config
      - cmd: activate-ssh-agent
      - pkg: packages_apache_php

#git@bitbucket.org:user/dokuwiki.git:
#  git.latest:
#    - user: root
#    - identity: "/root/.ssh/id_rsa"
#    - target: /var/www/html/wiki
#     - require:
#        - /root/.ssh/id_rsa
#        - /root/.ssh/id_rsa.pub
#        - /root/.ssh/known_hosts
```

**Comment by ssgward** [comment]
@gidantribal - so it looks like you got things working. Or am I misunderstanding?

**Comment by gidantribal** [comment]
@ssgward actually the raw `cmd.run: git` command works as expected, but I can't use the `salt.state.git `function (that is much cleaner and generic, of course)

**Comment by ssgward** [comment]
@gidantribal - great. Thanks. So it looks like the real issue here, and the fix that is being requested, is using a port other than port 22 if so specified when using salt.state.git

**Comment by gidantribal** [comment]
@ssgward I think there are two issues here, one is about being able to configure the port at `salt.state.git`, the second is that `salt.state.git` ignores any configuration stored in `.ssh/config` file (and it shouldn't be the case, because the `ssh` session used at `git` command normally takes into account any `ssh` configuration.

**Comment by rallytime** [comment]
FYI @terminalmage

**Comment by terminalmage** [comment]
You seem to have skipped providing any of the information from the issue template. Please let me know what version of Salt is running on your minion.

**Comment by terminalmage** [comment]
After some investigation, I can reproduce this, but only when manually specifying an identity file in the state. As a workaround, you can add an ``IdentityFile`` entry in your root ssh config file (``/root/.ssh/config``), pointing at ``/root/.ssh/id_rsa``, like so:

```
Host altssh.bitbucket.org
    Port 443
    IdentityFile /root/.ssh/id_rsa
```

Then, remove the ``identity`` param from your ``git.latest`` state.

I should be able to get a fix in for this next week after I return from vacation.

**Comment by gidantribal** [comment]
great, thank you!!

**Comment by gidantribal** [comment]
salt 2016.3.0rc2 (Boron)
salt-minion 2016.3.0rc2 (Boron)

**Comment by terminalmage** [comment]
Fixed in #33152.
